py_hamt/store.py

```
@@ -1,205 +1,191 @@
 
-    The return type of save and input to load is really type IPLDKind, but the documentation generates something a bit strange since IPLDKind is a type union.
+    #### A note on the IPLDKind return types
+    Save and load return the type IPLDKind and not just a CID. As long as python regards the underlying type as immutable it can be used, allowing for more flexibility. There are two exceptions:
+    1. No lists or dicts, since python does not classify these as immutable.
+    2. No `None` values since this is used in HAMT's `__init__` to indicate that an empty HAMT needs to be initialized.
     """
 
-    @abstractmethod
-    def save_raw(self, data: bytes) -> IPLDKind:
-        """Take any set of bytes, save it to the storage mechanism, and return an ID in the type of IPLDKind that can be used to retrieve those bytes later."""
+    type CodecInput = Literal["raw", "dag-cbor"]
```

Pylint (3.3.7)

[error] Parsing failed: 'invalid syntax (py_hamt.store, line 23)'

(E0001)